api/redhatopenshift/Hypershift/hcpCluster-models.tsp

```diff
@@ -0,0 +1,551 @@
+import "@typespec/rest";
+import "@typespec/http";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+
+using TypeSpec.Rest;
+using TypeSpec.Http;
+using Azure.Core;
+using Azure.ResourceManager;
+
+namespace Microsoft.RedHatOpenshift;
+
+/*
+* ===================================
+*   HyperShift cluster core resources
+* ===================================
+*/
+
+/** HyperShift cluster resource */
+model HcpOpenShiftClusterResource is TrackedResource<HcpOpenShiftClusterProperties> {
+  /** Name of HyperShift cluster */
+  @pattern("^[a-zA-Z0-9-]{3,24}$")
+  @minLength(3)
+  @maxLength(24)
+  @key("hcpOpenShiftClusterName") // sets the alternative name for the name propery
+  @path
+  @segment("hcpOpenShiftClusters")
+  name: string;
+
+  ...ManagedServiceIdentity;
+}
+
+// The NodePool needs to be TrackedResource for the following reasons:
+// - allow tags to be in place, which allows billing to use tags, important for workers
+// - allow nodepool to be tracked in the portal
+// - deleted when resource group is deleted
+// - allow cascade delete
+// more: https://armwiki.azurewebsites.net/rp_onboarding/tracked_vs_proxy_resources.html
+@parentResource(HcpOpenShiftClusterResource)
+model HcpOpenShiftClusterNodePoolResource is TrackedResource<NodePoolProperties> {
+  /** Name of HyperShift cluster */
+  @pattern("^[a-zA-Z0-9-]{3,24}$")
+  @minLength(3)
+  @maxLength(24)
+  @key("nodePoolName")
+  @path
+  @segment("nodePools")
+  name: string;
+}
+
+/** HyperShift cluster properties */
+model HcpOpenShiftClusterProperties {
+  /** The status of the last operation. */
+  @visibility("read")
+  provisioningState?: ProvisioningState;
+
+  /** The cluster resouce specification. */
+  @visibility("create", "update")
+  spec: ClusterSpec;
+}
+
+/** The cluster resource specification */
+model ClusterSpec {
+  /** Version of the control plane components */
+  @visibility("create", "update")
+  version: VersionProfile;
+
+  /** Cluster DNS configuration */
+  dns?: DnsProfile;
+
+  /** Cluster network configuration */
+  @visibility("create")
+  network?: NetworkProfile;
+
+  /** Shows the cluster web console information */
+  @visibility("read")
+  console: ConsoleProfile;
+
+  /** Shows the cluster API server profile */
+  @visibility("read")
+  api: ApiProfile;
+
+  /** Enable FIPS mode for the cluster
+   * When set to true, `etcdEncryption` must be set to true
+  */
+  @visibility("create")
+  fips?: boolean = false;
+
+  /** Enables customer ETCD encryption, set during creation
+   * When set to true, `platform.etcdEncryptionSetId` must be set
+  */
+  @visibility("create")
+  etcdEncryption?: boolean = false;
+
+  /** Disable user workload monitoring */
+  @visibility("create", "update")
+  disableUserWorkloadMonitoring?: boolean = false;
```

Comment:
```suggestion
  /** Disable user workload monitoring */
  @visibility("create", "update")
  enableUserWorkloadMonitoring?: boolean = true;
```
We'd like to rename this field in our ARM spec to avoid double-negatives/maintain consistency. Is this something that we can feed into an OCM API change eventually?